safekeeper/src/timeline.rs

```diff
@@ -777,6 +805,16 @@ impl GlobalTimelines {
             .collect()
     }
 
+    /// Return FullTimelineInfo for all active timelines.
+    pub fn active_timelines_metrics() -> Vec<FullTimelineInfo> {
+        let state = TIMELINES_STATE.lock().unwrap();
```

Comment:
Holding both map lock and per-tli lock during the whole scrape isn't cool and it is easy to avoid, but it is ok to see the real problem first.

Reply:
Right, I just did the same as in `get_active_timelines`. If scrape will take a substantial time (100ms+) then we'll get back to it.